spring-cloud-stream-binder-rocketmq/src/main/java/com/alibaba/cloud/stream/binder/rocketmq/consuming/RocketMQListenerBindingContainer.java

```diff
@@ -394,18 +399,20 @@ public ConsumeConcurrentlyStatus consumeMessage(List<MessageExt> msgs,
 
 	public class DefaultMessageListenerOrderly implements MessageListenerOrderly {
 
-		@SuppressWarnings("unchecked")
+		@SuppressWarnings({ "unchecked", "Duplicates" })
 		@Override
 		public ConsumeOrderlyStatus consumeMessage(List<MessageExt> msgs,
 				ConsumeOrderlyContext context) {
 			for (MessageExt messageExt : msgs) {
 				log.debug("received msg: {}", messageExt);
 				try {
 					long now = System.currentTimeMillis();
+					preOnMessage(messageExt);
```

Comment:
i think create a new method is better, it include `RocketMQUtil.convertToSpringMessage(messageExt)` & the logic of `preOnMessage(messageExt)`

Reply:
Good idea,this method is used to convert rocketmq message to spring message。